pontoon/sync/tasks.py

```diff
@@ -396,10 +378,12 @@ def sync_translations(
             "has anything to sync.".format(db_project.slug)
         )
 
-    for r in repos:
-        r.set_last_synced_revisions(
-            locales=repo_locales[r.pk].exclude(code__in=failed_locales)
-        )
+    if repo_locales:
+        repos = db_project.repositories.filter(pk__in=repo_locales.keys())
+        for r in repos:
+            r.set_last_synced_revisions(
+                locales=repo_locales[r.pk].exclude(code__in=failed_locales)
+            )
```

Comment:
I think you can also keep the original code here, if you set the last synced revision for the source repository separately.

Reply:
This comment is now also obsolete.